app/models/manageiq/providers/cloud_manager/virtual_template.rb

```diff
@@ -0,0 +1,14 @@
+class ManageIQ::Providers::CloudManager::VirtualTemplate < ::MiqTemplate
+  validate :single_template, :on => :create
+  default_value_for :cloud, true
+
+  TYPES = {
+    :amazon => 'ManageIQ::Providers::Amazon::CloudManager::VirtualTemplate'
```

Comment:
@Fryguy yes, was going to work on this. This was just a shell of what the models were to be - since my priority here was to get the API done. Will talk with @durandom